perses/app/relative_point_mutation_setup.py

```diff
@@ -190,40 +205,46 @@ def __init__(self,
-                                                                 chain_id = mutation_chain_id, #denote the chain id allowed to mutate (it's always a string variable)
-                                                                 max_point_mutants = 1,
-                                                                 residues_allowed_to_mutate = [mutation_residue_id], #the residue ids allowed to mutate
-                                                                 allowed_mutations = [(mutation_residue_id, proposed_residue)], #the residue ids allowed to mutate with the three-letter code allowed to change
-                                                                 aggregate = True) #always allow aggregation
+        for (top, pos, sys) in inputs:
+            point_mutation_engine = PointMutationEngine(wildtype_topology=top,
+                                                                 system_generator=self.system_generator,
+                                                                 chain_id=mutation_chain_id, # Denote the chain id allowed to mutate (it's always a string variable)
+                                                                 max_point_mutants=1,
+                                                                 residues_allowed_to_mutate=[mutation_residue_id], # The residue ids allowed to mutate
+                                                                 allowed_mutations=[(mutation_residue_id, proposed_residue)], # The residue ids allowed to mutate with the three-letter code allowed to change
+                                                                 aggregate=True) # Always allow aggregation
 
             topology_proposal = point_mutation_engine.propose(sys, top)
 
-            new_positions, logp_proposal = geometry_engine.propose(topology_proposal, pos, beta)
-            logp_reverse = geometry_engine.logp_reverse(topology_proposal, new_positions, pos, beta)
+            # Only validate energy bookkeeping if the WT and proposed residues do not involve rings
+            old_res = [res for res in top.residues() if res.id == mutation_residue_id][0]
+            validate_bool = False if old_res in ring_amino_acids or proposed_residue in ring_amino_acids else True
 
-            forward_htf = HybridTopologyFactory(topology_proposal = topology_proposal,
-                                                 current_positions =  pos,
-                                                 new_positions = new_positions,
-                                                 use_dispersion_correction = False,
+            new_positions, logp_proposal = geometry_engine.propose(topology_proposal, pos, beta,
+                                                                   validate_energy_bookkeeping=validate_bool)
+            logp_reverse = geometry_engine.logp_reverse(topology_proposal, new_positions, pos, beta,
```

Comment:
this is defined, but never used

Reply:
in the future, presumably rjmc will be used...it also lives in `RelativeFEPSetup`